modules/code-editor/src/views/full/FileNavigator.tsx

```
@@ -77,14 +83,87 @@ export default class FileNavigator extends React.Component<Props, State> {
     }
   }
 
+  handleContextMenu = (node: ITreeNode<NodeData>, path, e) => {
+    if (!node.nodeData) {
+      return null
+    }
+
+    e.preventDefault()
+    ContextMenu.show(
+      <Menu>
+        <MenuItem icon="edit" text="Rename" onClick={() => this.handleRename(node)} />
+        <MenuItem icon="delete" text="Delete" onClick={() => this.props.onNodeDelete(node.nodeData as EditableFile)} />
+      </Menu>,
+      { left: e.clientX, top: e.clientY }
+    )
+  }
+
+  handleRename = (node: ITreeNode<NodeData>) => {
+    const nodeDomElement = this.treeRef.current.getNodeContentElement(node.id)
+
+    const input = document.createElement('input')
+    input.type = 'text'
+    input.className = 'bp3-input bp3-small'
+    input.value = node.label as string
+
+    const div = document.createElement('div')
+    div.className = nodeDomElement.className
+    div.appendChild(input)
+
+    nodeDomElement.replaceWith(div)
+
+    input.focus()
+    input.select()
+
+    const closeRename = async e => {
+      e.preventDefault()
+      div.replaceWith(nodeDomElement)
+      window.removeEventListener('keydown', keyboardListener)
+      window.removeEventListener('mousedown', mouseListener)
+
+      const file = node.nodeData as EditableFile
+      let newName = input.value as string
+      newName = newName.endsWith('.js') ? newName : newName + '.js'
+
+      if (newName === node.label) {
+        return
+      }
+
+      try {
+        await this.props.onNodeRename(file, newName)
+      } catch (e) {
+        return
+      }
```

It's never a good thing to catch silently errors, it can make debugging hard. Always at least log to the console (even better: use a toast, since you want to know why it failed). If it should really be silent, add a comment explaining why